akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -317,6 +318,96 @@ class CircuitBreaker(
   }
 
   /**
+   * Adds a callback to execute when call finished with success.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Long ⇒ Unit): CircuitBreaker = onCallSuccess(new Consumer[Long] {
+    def accept(result: Long): Unit = callback(result)
+  })
+
+  /**
+   * JavaAPI for onCallSuccess
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Consumer[Long]): CircuitBreaker = {
```

Comment:
Inventing different names is the only way, but it's so difficult to come up with good names for such things. The best would be to only have one signature, and that would work for 2.12, but then we would have to maintain different sources for 2.11 and 2.12